src/FileIO/FileIO.js

```diff
@@ -0,0 +1,68 @@
+/*******************************************************************************
+*  Copyright by the contributors to the Dafny Project
+*  SPDX-License-Identifier: MIT
+*******************************************************************************/
+
+var DafnyLibraries = DafnyLibraries || {};
+DafnyLibraries.FileIO = (function() {
+  const buffer = require("buffer");
+  const fs = require("fs");
+  const nodePath = require("path");
+
+  let $module = {};
+
+  /**
+   * Attempts to read all bytes from the file at the given `path`, and returns an array of the following values:
+   *
+   *   - `isError`: true iff an error was thrown during path string conversion or when reading the file
+   *   - `bytesRead`: the sequence of bytes from the file, or an empty sequence if `isError` is true
+   *   - `errorMsg`: the error message of the thrown error if `isError` is true, or an empty sequence otherwise
+   *
+   * We return these values individually because `Result` is not defined in the runtime but instead in library code.
+   * It is the responsibility of library code to construct an equivalent `Result` value.
+   */
+  $module.INTERNAL_ReadBytesFromFile = function(path) {
+    const emptySeq = _dafny.Seq.of();
+    try {
+      const readOpts = { encoding: null };  // read as buffer, not string
+      const buf = fs.readFileSync(path, readOpts);
+      const readBytes = _dafny.Seq.from(buf.valueOf(), byte => new BigNumber(byte));
```

Comment:
Oof, we definitely have to optimize the crap out of the JS backend for byte sequences later :P